src/hooks/useNetworkSwitch.ts

```diff
@@ -17,25 +18,50 @@ export type UseNetworkSwitchProps = {
 export const useNetworkSwitch = ({ onSelectNetworkCallback }: UseNetworkSwitchProps = {}) => {
   const { connector, chainId, account } = useActiveWeb3React()
   const unsupportedChainIdError = useUnsupportedChainIdError()
+  const navigate = useNavigate()
+  const { pathname } = useLocation()
 
   const selectNetwork = useCallback(
-    (optionChainId?: ChainId) => {
+    async (optionChainId?: ChainId) => {
       if (optionChainId === undefined || (optionChainId === chainId && !unsupportedChainIdError)) return
+
+      let changeChainId: Promise<unknown> | undefined
```

Comment:
Since this is a promise I think we should have a different name for this variable. Maybe something related with `result` | `response`.

Reply:
Well, it's not exactly a result or a response but I can give it a more representative name... maybe with the word `promise included`